Agents/PubChemAgent/pubchemagent/app.py

```diff
@@ -0,0 +1,101 @@
+from pubchemagent.kgoperations.getkgdata import *
+from pubchemagent.kgoperations.addkgdata import * 
+from pubchemagent.pug import pug_api, chebi_api 
+from pubchemagent.pug.chebi_classification import get_chebi_tree
+import time
+import logging
+
+logging.getLogger('urllib3').setLevel(logging.CRITICAL)
+logging.getLogger('requests').setLevel(logging.CRITICAL)
+logging.getLogger('requests_cache').setLevel(logging.CRITICAL)
+logging.getLogger('bioservices').setLevel(logging.CRITICAL)
+logging.getLogger('matplotlib').setLevel(logging.CRITICAL)
+logging.getLogger('suds').setLevel(logging.CRITICAL)
+
+def species_instantiation(inchi):
+    pug_access = pug_api()
+    start_time = time.time()
+    IRI, flag = get_iri_data(inchi)
+    if IRI:
+        uuid = IRI.partition('_')[2]
+    else:
+        uuid = create_uuid()
+        flag = "true"
+    print("--- Query IRI: %s seconds ---" % (time.time() - start_time))
+
+    if flag == "true":
+        start_time = time.time()
+        data = pug_access.pug_request('InChI', inchi)
+        cid, props, identifiers = pug_access.get_props(data)
+        cid = pug_access.check_preferred(cid)
+        data_3d = pug_access.pug_request_prop_3d(cid)
+        if 'Record' in data_3d:
+            geometry, bonds = pug_access.get_structure('3d', data_3d)
+        else:
+            geometry, bonds = pug_access.get_structure('2d', data)
+        synonyms = pug_access.pug_request_synonyms(cid)
+        exp_props = pug_access.pug_request_exp_prop(cid)
+        uses = pug_access.pug_request_uses(cid)
+        ghs = pug_access.pug_request_ghs_classification(cid)
+        spectra = pug_access.pug_request_spectra(cid)
+        chebi_prop = chebi_api.chebi_request(inchi)
+        print("--- Import PubChem: %s seconds ---" % (time.time() - start_time))
+
+        for item in identifiers:
+            if identifiers[item]['key'] == 'MolecularFormula':
+                MolecularFormula = identifiers[item]['value']
+
+        typeIRI = '<http://www.theworldavatar.com/ontology/ontospecies/OntoSpecies.owl#Species>'  
```

Comment:
You've got numerous hard-coded URL strings like this one all over the code. This makes the code very hard to maintain (imagine if you have to change the base URL, then you've got one heck of a find-and-replace job on your hands...). If it is not realistic to clean this up at this stage, then please add a to-do section in your main readme file that this needs to be (urgently) cleaned up.

Reply:
I removed all the hard-coded URL. The URL are now defined in utils/url_configs.py file